binderhub/static/index.js

```diff
@@ -102,7 +104,7 @@ function updateUrl() {
   repo = repo.replace(/^(https?:\/\/)?github.com\//, '');
   // trim trailing or leading '/' on repo
   repo = repo.replace(/(^\/)|(\/?$)/g, '');
-  var ref = $('#ref').val().trim();
+  var ref = $('#ref').val().trim() || 'master';
```

Comment:
empty strings aren't falsy in javascript, so `|| master` will never run. We'd have to add

```javascript
if (ref.length === 0) ref = 'master';
```

to get this behavior.

Reply:
`||` seem to work for me.... but happy to change. 
<img width="175" alt="screen shot 2017-10-20 at 11 20 56" src="https://user-images.githubusercontent.com/335567/31836052-ca4d170e-b588-11e7-84a7-ad2e8e40ce1f.png">

Reply:
[According to MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators), empty-string is casted to false.

Reply:
Whoa, not sure where I'm getting my info. Sorry for misleading!